process/sync/shardForkDetector.go

```diff
@@ -82,8 +82,8 @@ func (sfd *shardForkDetector) AddHeader(
 func (sfd *shardForkDetector) addFinalHeaders(finalHeaders []data.HeaderHandler, finalHeadersHashes [][]byte) {
 	finalCheckpointWasSet := false
 	for i := 0; i < len(finalHeaders); i++ {
-		isFinalHeaderNonceHigherThanCurrent := finalHeaders[i].GetNonce() > sfd.GetHighestFinalBlockNonce()
-		if isFinalHeaderNonceHigherThanCurrent {
+		isFinalHeaderNonceHigherThanGenesis := finalHeaders[i].GetNonce() > process.GenesisBlockNonce
+		if isFinalHeaderNonceHigherThanGenesis {
 			if !finalCheckpointWasSet {
 				sfd.setFinalCheckpoint(&checkpointInfo{nonce: finalHeaders[i].GetNonce(), round: finalHeaders[i].GetRound()})
```

Comment:
It does not make sense to set a final checkpoint with nonce lower than current final nonce.

Reply:
There would be no such case, but with the old code there was a problem when a node rolled back and reapply a final header list. Each time the final header will increase with one because of the check.